fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1594,6 +1597,32 @@ ACTOR Future<Void> postResolution(CommitBatchContext* self) {
 		                        &self->computeStart));
 	}
 
+	// Add idempotency ids
```

Comment:
Can we add a knob to enable or disable this feature, since it's on the critical path?

Reply:
It's already off by default and can be enabled/disabled since you need to set a new transaction option. It's not clear to me what we should do here if we want to disable it server-side, but clients are sending requests with idempotency ids, since if we ignore it and the client is relying on it then that could result in corrupt data.

Reply:
Even it's off by default, this loop is still doing a lot of work, which is undesirable. Before rolling out this feature, we probably don't want to pay the cost, and a knob would be desirable and useful to quantify the performance impact of this feature.

Regarding to how the client behaves if the server side doesn't support it, one idea is to let the client know when constructing the `Database` object. If the new transaction option is used, then an error could be immediately thrown on the client side.

Reply:
I added a benchmark which shows that the cost when no idempotency ids are valid is about 4ns per transaction, which we agreed offline is small.